backend/hitas/views/apartment.py

```diff
@@ -922,6 +922,18 @@ def get_detail_queryset(self):
             sapc=self.select_sapc(calculation_month=calculation_month),
         )
 
+    @action(detail=True, methods=["GET"], url_path="retrieve-unconfirmed-prices-for-date")
+    def retrieve_unconfirmed_prices_for_date(self, request, **kwargs):
+        input_data = UnconfirmedPriceSerializer(data=request.query_params)
```

Comment:
Shouldn't this fetch from confirmed, not unconfirmed max price calculations?

Reply:
It might make sense to show on UI also what affects what somehow. And I suspect the same the confirmed price should be searched from the date chosen for sale date and if not available the limitprice from that date used.

Reply:
Nope, it is intentionally using unconfirmed prices, as it should be able to use the surface area price ceiling for past sales also. Frontend uses the latest confirmed calculation if it is valid on the sale date, if we need to use an no-most-recent confirmed calculation thats a new feature (also commented this on the jira ticket)

Reply:
Discussed this with the customer, and they said that a new confirmed max price calculation should be made for the given date if one does not already exist since re-sales should always be made using them according to Hitas rules. Therefore we changed the apartment endpoint to fetch the most recent max price calculation before a given date, which also fixes a small issue in the apartment view that would show the most recently confirmed max price, even if there is one which is valid for longer. Frontend changes are still required to make this work.